### PR TITLE
[AC-10072] : Outlook calendar link shows time different from the specified timezone

### PR DESCRIPTION
**Pull request by AmosWels**
#### What does this PR do?
- Outlook calendar link shows time different from the specified timezone

#### Description of Task to be completed?
- Outlook calendar link shows time different from the specified timezone

#### How should the reviewer navigate to the affected pages? (if appropriate)
- As a finalist, navigate to http://localhost:8181/officehours/ and reserve an office hour. Check the console for the emails sent and confirm that the def of done according to [AC-10072](https://masschallenge.atlassian.net/browse/AC-10072) is achieved

#### Any extra settings needed to be configured in the admin interface?
- `git checkout AC-10072 & git pull`
- `make build`
- `make run-all-servers && docker attach web-accelerate-container`
- `make bash-shell`
- Run `python3` command and execute the steps below
> from datetime import datetime, timedelta
> from pytz import utc, timezone
> from add2cal import Add2Cal
> title = "test title new"
> description = "description new"
> location = `"https://masschallenge.zoom.us/j/89590985719?from=addon"`
> timezone = "Africa/Lagos"
> date = utc.localize(datetime(2023,10,6,10))
> from add2cal.add2cal import INPUT_DATE_FORMAT
> start = date.strftime(INPUT_DATE_FORMAT)
> end = (date + timedelta(minutes=30)).strftime(INPUT_DATE_FORMAT)
> a2c = Add2Cal(start=start, end=end, title=title, description=description, location=location, timezone=timezone)

- `a2c.outlook_calendar_url()` _// outlook calendar link_
- `a2c.google_calendar_url()` _//google calendar link_
- `a2c.yahoo_calendar_url()` _//yahoo calendar link_

#### Any background context you want to provide?
- Outlook calendar wasn't updating the expected user timezone provided by the user

#### What are the relevant Jira stories?
- [AC-10072](https://masschallenge.atlassian.net/browse/AC-10072)

#### Screenshots (if appropriate)

[Sibling PR](https://github.com/masschallenge/accelerate/pull/3725)

[AC-10072]: https://masschallenge.atlassian.net/browse/AC-10072?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AC-10072]: https://masschallenge.atlassian.net/browse/AC-10072?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: add2cal/add2cal.py [hunk 1]
```diff
@@ -11,12 +11,12 @@
...
 BASE_URLS = {
     'google': 'https://calendar.google.com/calendar/render',
-    'outlook': 'https://outlook.office.com/owa/',
+    'outlook': 'https://outlook.office.com/calendar/action/compose',
     'yahoo': 'http://calendar.yahoo.com'
 }
 
 INPUT_DATE_FORMAT = "%Y%m%dT%H%M%S"
-OUTLOOK_DATE_FORMAT = '%Y-%m-%dT%I:%M:%SZ'
+OUTLOOK_DATE_FORMAT = '%Y-%m-%dT%I:%M:%S'
```
Comment (iankigen) [inline add2cal/add2cal.py:19]: why remove the timezone info?

**Review by iankigen — COMMENTED** [review]
From QA
- Outlook calendar info is incorrect.

Example: I created and reserved an OH session for 17th Sept (Sunday) 
Time 1:00 pm - 2:00 pm EAT (+3 GMT) 

```
{
...
 'id': 51984,
 'created_at': datetime.datetime(2023, 9, 13, 18, 15, 14, tzinfo=<UTC>),
 'updated_at': datetime.datetime(2023, 9, 13, 18, 20, 34, tzinfo=<UTC>),
 'program_id': None,
 'mentor_id': 78850,
 'finalist_id': 78851,
 'startup_id': None,
 'start_date_time': datetime.datetime(2023, 9, 17, 10, 0, tzinfo=<UTC>),
 'end_date_time': datetime.datetime(2023, 9, 17, 11, 0, tzinfo=<UTC>),
 'description': 'Hello world',
 ...}
 ```
 API Response
 ```
 {
    "success": true,
    "header": "Office Hour reserved with ian Kigen",
    "detail": "",
    "timecard_info": {
        "finalist_first_name": "Ben",
        "finalist_last_name": "Mark",
        "finalist_email": "[ben@supportomate.com](mailto:ben@supportomate.com)",
        "topics": "",
        "startup": "",
        "calendar_data": {
            "outlook_link": "https://outlook.office.com/calendar/action/compose?rru=addevent&startdt=2023-09-17T01%3A00%3A00&enddt=2023-09-17T02%3A00%3A00&subject=Office+Hours+Session+with+ikigen%40masschallenge.org&uid=5bc0b3a3fba935511f08ede1d79575be&location=Jerusalem+%7C+Israel+Remote%3Bzoom.us%2Ftest&body=%0A++++++++%0A++++++++Attendees%3A%0A-+ikigen%40masschallenge.org%0A-+ben%40supportomate.com+-+%0A%0A++++++++%0A%0A++++++++%0A++++++++&allday=",
            "gcal_link": "https://calendar.google.com/calendar/render?action=TEMPLATE&text=Office+Hours+Session+with+ikigen%40masschallenge.org&dates=20230917T130000%2F20230917T140000&details=%0A++++++++%0A++++++++Attendees%3A%0A-+ikigen%40masschallenge.org%0A-+ben%40supportomate.com+-+%0A%0A++++++++%0A%0A++++++++%0A++++++++&location=Jerusalem+%7C+Israel+Remote%3Bzoom.us%2Ftest&pli=1&uid=5bc0b3a3fba935511f08ede1d79575be&sf=true&ctz=Africa%2FNairobi&output=xml&followup=https%3A%2F%2Fcalendar.google.com%2Fcalendar%2F&scc=1&authuser=0",
            "yahoo_link": "[http://calendar.yahoo.com?v=60&view=d&type=20&uid=&title=Office+Hours+Session+with+ikigen%40masschallenge.org&st=20230917T130000&ctz=Africa%2FNairobi&in_loc=Jerusalem+%7C+Israel+Remote%3Bzoom.us%2Ftest&dur=0100&desc=%0A++++++++%0A++++++++Attendees%3A%0A-+ikigen%40masschallenge.org%0A-+ben%40supportomate.com+-+%0A%0A++++++++%0A%0A++++++++%0A++++++++](http://calendar.yahoo.com/?v=60&view=d&type=20&uid=&title=Office+Hours+Session+with+ikigen%40masschallenge.org&st=20230917T130000&ctz=Africa%2FNairobi&in_loc=Jerusalem+%7C+Israel+Remote%3Bzoom.us%2Ftest&dur=0100&desc=%0A++++++++%0A++++++++Attendees%3A%0A-+ikigen%40masschallenge.org%0A-+ben%40supportomate.com+-+%0A%0A++++++++%0A%0A++++++++%0A++++++++)",
            "ical_content": "BEGIN:VCALENDAR\r\nPRODID:ics.py - [http://git.io/lLljaA\r\nVERSION:2.0\r\nBEGIN:VEVENT\r\nDTSTAMP:20230913T182033\r\nDTSTART;TZID=Africa/Nairobi:20230917T130000\r\nDTEND;TZID=Africa/Nairobi:20230917T140000\r\nSUMMARY:Office](http://git.io/lLljaA%5Cr%5CnVERSION:2.0%5Cr%5CnBEGIN:VEVENT%5Cr%5CnDTSTAMP:20230913T182033%5Cr%5CnDTSTART;TZID=Africa/Nairobi:20230917T130000%5Cr%5CnDTEND;TZID=Africa/Nairobi:20230917T140000%5Cr%5CnSUMMARY:Office) Hours Session with ikigen@masschallenge.org\r\nDESCRIPTION:\\n        \\n        Attendees:\\n- [ikigen@masschallenge.org\\n-](mailto:ikigen@masschallenge.org\\n-) [ben@supportomate.com](mailto:ben@supportomate.com) - \\n\\n        \\n\\n        \\n        \r\nLOCATION:Jerusalem | Israel Remote\\;zoom.us/test\r\nTRANSP:OPAQUE\r\nUID:944d3b2c-e414-4750-8461-9f38369ae30b@944d.org\r\nBEGIN:VALARM\r\nTRIGGER;VALUE=DATE-TIME:20230917T010000Z\r\nACTION:DISPLAY\r\nDESCRIPTION:\r\nEND:VALARM\r\nEND:VEVENT\r\nEND:VCALENDAR"
        }
    }
}
 ```
 
 - Google Calendar ✅ 
 - 
Screenshot 2023-09-13 at 21 27 45

 - Outlook Calendar ❌ 
 - 
Screenshot 2023-09-13 at 21 28 24


Outlook calendar is lacking timezone information and is displaying incorrect start/end time
i.e 1:00 AM to 2:00 AM instead of 1:00 pm - 2:00 pm EAT (+3 GMT)

File: add2cal/add2cal.py (hunk 1 above)
Comment (AmosWels) [reply add2cal/add2cal.py]: we have the dates already in the user's preferred timezone is available or UTC. If added, the outlook timezone displays a different unexpected time

**Comment by AmosWels** [comment]
> 12:00 PM - 1:00 PM

@iankigen , I've just created an office hour 5:30 pm-6:00 pm UTC+0 and a finalist in UTC +3 reserves and receives the calendar links with office hour time 8:30pm to 9:00pm

**Comment by kenware** [comment]
@AmosWels this is the two calendar I am getting:
Screenshot 2023-09-20 at 19 25 19

GOOGLE
Screenshot 2023-09-20 at 19 24 17

OUTLOOK
Screenshot 2023-09-20 at 19 24 39

The startdate and enddata seems different for google calendar and outlook

**Review by AmosWels — COMMENTED** [review]
From the current commit, we now have timezone conversion only on accelerate. The time a user sees on accelerate will be the same time returned on all the three calendars.

- We have the date converted to the user timezone as we know on the user-preffered timezone on accelerate
- We now send already formatted dates to the calendar
- No date timezone conversion happens on any of the calendar.

**How to Test** 
- Reserve office hours, get the email sent to the mentor and finalist, open in browser, and create the event.
- The time showing on office hours should be the same as the one displayed on all three calendars as indicated in the timezone.

**Review by kenware — APPROVED** [review]
Looks good

**Review by iankigen — APPROVED** [review]

**Pull request merged**